org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -548,7 +548,7 @@ public void didChangeConfiguration(DidChangeConfigurationParams params) {
 		if (settings instanceof Map) {
 			Collection<IPath> rootPaths = preferenceManager.getPreferences().getRootPaths();
 			@SuppressWarnings("unchecked")
-			Preferences prefs = Preferences.createFrom((Map<String, Object>) settings);
+			Preferences prefs = Preferences.updateFrom(preferenceManager.getPreferences(), (Map<String, Object>) settings);
```

Comment:
Is it worth doing the same for : https://github.com/eclipse-jdtls/eclipse.jdt.ls/blob/6a1bd24e69e581f8c927c87c9064ca611b2bfd09/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/syntaxserver/SyntaxLanguageServer.java#L276

Reply:
fixed